rest/blip_sync.go

```diff
@@ -71,3 +79,31 @@ func (h *handler) handleBLIPSync() error {
 
 	return nil
 }
+
+func (sc *ServerContext) limitConcurrentReplications(ctx context.Context) (func(sc *ServerContext), error) {
+	if sc.Config.Replicator.MaxConcurrentConnections == 0 {
+		return nil, nil
+	}
+	sc.lock.Lock()
+	defer sc.lock.Unlock()
+
+	capacity := sc.Config.Replicator.MaxConcurrentConnections
+	count := sc.activeReplicatorCount
+
+	release := func(sc *ServerContext) {
+		sc.lock.Lock()
+		defer sc.lock.Unlock()
+		connections := sc.Config.Replicator.MaxConcurrentConnections
+		sc.activeReplicatorCount--
+		base.TracefCtx(ctx, base.KeyHTTP, "Released replication slot (remaining: %d/%d)", connections-sc.activeReplicatorCount, connections)
+	}
+
+	if count >= capacity {
+		base.InfofCtx(ctx, base.KeyHTTP, "Replication limit exceeded (active: %d)", count)
+		return nil, base.ErrReplicationLimitExceeded
+	} else if count < capacity {
+		sc.activeReplicatorCount++
+		base.TracefCtx(ctx, base.KeyHTTP, "Acquired replication slot (remaining: %d/%d)", capacity-sc.activeReplicatorCount, capacity)
```

Comment:
I'd probably log this as 'active: %d/%d' since it provides the same information and avoids evaluation of `capacity - sc.activeReplicatorCount`, which would occur even if trace logging isn't enabled.